src/test/scala/com/bazaarvoice/sswf/WaitTimeTest.scala

```diff
@@ -0,0 +1,154 @@
+package com.bazaarvoice.sswf
+
+import com.amazonaws.services.simpleworkflow.AmazonSimpleWorkflowClient
+import com.amazonaws.services.simpleworkflow.model._
+import com.bazaarvoice.sswf.model.history.StepsHistory
+import com.bazaarvoice.sswf.model.result.{InProgress, StepResult, Success}
+import com.bazaarvoice.sswf.model.{DefinedStep, ScheduledStep, StepInput}
+import com.bazaarvoice.sswf.service.{StepActionWorker, StepDecisionWorker, WorkflowManagement}
+import example.StdOutLogger
+import org.joda.time.Duration
+import org.scalatest.FlatSpec
+
+import scala.collection.JavaConversions._
+
+class WaitTimeTestWorkflowDef() extends WorkflowDefinition[String, WaitTimeSteps] {
+  private var waitStepInvocationCount: Int = 4
+  override def workflow(input: String): _root_.java.util.List[ScheduledStep[WaitTimeSteps]] = List(DefinedStep(WaitTimeSteps.DUMMY_STEP), DefinedStep(WaitTimeSteps.WAIT_STEP), DefinedStep
+  (WaitTimeSteps.DUMMY_STEP))
+  override def onFinish(workflowId: String, runId: String, input: String, history: StepsHistory[String, WaitTimeSteps], message: String): Unit = {}
+  override def onCancel(workflowId: String, runId: String, input: String, history: StepsHistory[String, WaitTimeSteps], message: String): Unit = {}
+  override def onFail(workflowId: String, runId: String, input: String, history: StepsHistory[String, WaitTimeSteps], message: String): Unit = {}
+
+  override def act(step: WaitTimeSteps, input: String, stepInput: StepInput, heartbeatCallback: HeartbeatCallback, execution: WorkflowExecution): StepResult =
+    step match {
+      case WaitTimeSteps.DUMMY_STEP =>
+        Thread.sleep(2000)
+        Success(None)
+      case WaitTimeSteps.WAIT_STEP  => if (waitStepInvocationCount > 0) {
+        Thread.sleep(2000)
+        waitStepInvocationCount -= 1
+        InProgress(Some(s"waiting!!"))
+      } else {
+        Success(None)
+      }
+    }
+}
+
+class WaitTimeTest extends FlatSpec {
+
+  val parser = new InputParser[String] {
+    override def serialize(input: String): String = input
+    override def deserialize(inputString: String): String = inputString
+  }
+  private val domain: String = "sswf-tests"
+  private val wf: String = "wait-test"
+  private val swf: AmazonSimpleWorkflowClient = new AmazonSimpleWorkflowClient()
+  private val logger: StdOutLogger = new StdOutLogger
+  val manager = new WorkflowManagement[String, WaitTimeSteps](domain, wf, "0.0", wf, swf, inputParser = parser, log = logger)
+  val definition = new WaitTimeTestWorkflowDef()
+  val actor = new StepActionWorker[String, WaitTimeSteps](domain, wf, swf, parser, definition, log = logger)
+  val decider = new StepDecisionWorker[String, WaitTimeSteps](domain, wf, swf, parser, definition, logger)
+
+  "Each Step" should " have 2 invocations " in {
+    manager.registerWorkflow()
+    val workflow: WorkflowExecution = manager.startWorkflow("B", "")
+    try {
```

Comment:
We also need these tests:

Verify cumulative computation and resetting:
```
A  STEP0: Success()      verify: A.cum >= A.end - A.start
B  STEP0: Success()      verify: B.cum >= B.end - B.start AND B.cum != 0 AND A.cum != 0 AND B.cum < B.end - A.start
```

Verify cumulative computation over successive invocations:
```
A  STEP1: InProgress()
B  STEP1: Success()      verify: B.cum >= B.end - A.start
```